src/main/java/core/mate/academy/model/TruckProducer.java

```diff
@@ -0,0 +1,16 @@
+package core.mate.academy.model;
+
+import core.mate.academy.service.MachineProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TruckProducer extends Truck implements MachineProducer {
```

Comment:
```suggestion
public class TruckProducer implements MachineProducer<Truck> {
```
we don't need to extend class Truck here, but have to paramatrize imlemented interface
same for two other classes